plasmapy/diagnostics/path_integrated_diagnostic.py

```diff
@@ -0,0 +1,627 @@
+
+        # Reshape the integrands from (nx*ny*nz) to (nx, ny, nz)
+        integrand = np.reshape(integrand, (nx, ny, nz))
+
+        return integrand
+
+    def evaluate(
+        self,
+        num,
+        size=None,
+        bins=[200, 200],
+        collimated=True,
+    ):
+
+        r"""
+        Evaluates the line integral through the
+        provided grid. This is accomplished through the following steps:
+
+        1. Create an array of points in the detector plane using the detector
+           location and the size and bins keywords.
+
+        2. For each cell of detector grid, create an array of points of
+           separation ds from there to the source point (or,
+           when collimated=True, the source plane).
+
+        3. Evaluate the integrand function at each point.
+
+        4. Integrate along the line-integrated dimension to obtain the
+           line-integrated quantity in the detector plane.
+
+        Parameters
+        ----------
+        num : int
+            Number of integration points along the line (within the grid region).
+
+        size : `~astropy.units.Quantity` array of shape [2,2], optional
+            The bounds of the detector region. By default the size will be
+            chosen to be slightly larger than the image of the grid on the
+            detector plane.
+
+        bins : list of two ints, [N,], optional
+            Number of bins in each direction of the detector region. The
+            default is [200, 200].
+
+        collimated : Boolean, optional
+            If True, the source will be assumed to be collimated. If False,
+            a point source will be used. The default is True (collimated).
+
+        Returns
+        -------
+        xax : `~astropy.units.Quantity` array (Nh,)
+            The horizontal axis of the detector plane
+
+        yax : `~astropy.units.Quantity` array of shape (Nv,)
+            The vertical axis of the detector plane
+
+        integral : `~astropy.units.Quantity` array or list of arrays of shape (Nh, Ny)
+           The line-integrated values in the detector plane.
+
+        """
+        return self._line_integral(
+            num,
+            size=size,
+            bins=bins,
+            collimated=collimated,
+        )
+
+
+class Interferometer(LineIntegrateScalarQuantities):
+    def __init__(self, grid, source, detector, verbose=False):
+        super().__init__(grid, source, detector, quantities="n_e", verbose=verbose)
+
+    def evaluate(
+        self,
+        probe_freq: u.Hz,
+        num,
+        size=None,
+        bins=[200, 200],
+        collimated=True,
+        unwrapped=True,
+    ):
+        r"""
+        Creates an interferogram by calculating the line integral through
+        the electron number density :math:`n_e` provided on the grid. The phase shift
+        is
+
+        .. math:: \Delta \phi = -\frac{\omega_{probe}}{2 c n_c} \int n_e dl
```

Comment:
If you don't want to make this  raw string, then you'll have to put an escape character on math functions; that is, 

```rst
 .. math:: \\Delta \\phi = -\\frac{\\omega_{probe}}{2 c n_c} \\int n_e dl
```

Reply:
Oh ok, done